pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -279,22 +277,40 @@ func GatherContainerImages(i *Gatherer) func() ([]record.Record, []error) {
 				if strings.HasPrefix(pod.Namespace, "openshift") && hasContainerInCrashloop(podPtr) {
 					records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pod.Namespace, pod.Name), Item: PodAnonymizer{podPtr}})
 				} else if pod.Status.Phase == corev1.PodRunning {
-					for _, container := range pod.Spec.Containers {
-						imageURL := container.Image
-						urlHost, err := forceParseURLHost(imageURL)
-						if err != nil {
-							klog.Errorf("unable to parse container image URL: %v", err)
-							continue
-						}
-						if imageHostRegex.MatchString(urlHost) {
-							imgIndex, ok := image2idx[imageURL]
-							if !ok {
-								imgIndex = contInfo.Images.Add(imageURL)
-								image2idx[imageURL] = imgIndex
+					gatherImages := func(containers []corev1.ContainerStatus) {
+						for _, container := range containers {
+							dockerRef, err := reference.Parse(container.Image)
+							if err != nil {
+								klog.Errorf("unable to parse container image specification: %v", err)
+								continue
+							}
+
+							// Use the sha256 hash ID if available, otherwise use the full image spec.
+							imgMinimal := dockerRef.ID
+							if imgMinimal == "" {
+								imgMinimal = container.Image
+							}
+
+							const yyyyMmFormat = "2006-01"
+							startMonth := pod.CreationTimestamp.Time.UTC().Format(yyyyMmFormat)
+
+							if countMap, ok := img2month2count[imgMinimal]; ok {
+								var count int
```

Comment:
nit: `count := 0`

Reply:
Is it really a better way of declaring a variable that doesn't need a known default value? Of course, it will be initialized to `0` by default anyways, but to me, this syntax better indicates that it's supposed to be set later in the code and that the initial value shouldn't be used.